packages/tools/devtools/devtools-core/src/data-visualization/DefaultVisualizers.ts

```diff
@@ -248,20 +250,27 @@ export const visualizeSharedTree: VisualizeSharedObject = async (
 ): Promise<FluidObjectTreeNode> => {
 	const sharedTree = sharedObject as ISharedTree;
 
-	const contextRoot = sharedTree.view.context.root;
+	const schema = treeSchemaFromStoredSchema(sharedTree.contentSnapshot().schema);
+	const view = sharedTree.requireSchema(schema, () => {
+		throw new Error("Schema changed");
+	});
+	if (view === undefined) {
+		throw new Error("failed to apply schema");
+	}
+	const contextRoot = view.editableTree2(schema);
+
 	const children: Record<string, VisualChildNode> = {};
 
-	const iterateNodes = async (field: UntypedField): Promise<void> => {
-		for (const child of field) {
-			await iterateFields(child as unknown as UntypedTree);
+	const iterateNodes = async (field: TreeField): Promise<void> => {
```

Comment:
the update to editable-tree2 could be split into a separate PR if that would ease things.